Remove `dest_channel_id` from NetworkChannel

LasseRosenow opened the pull request:
I think it is more reasonable to not duplicate this value if we anyways for example have the ports or file descriptors stored in the concrete channel implementation structs.

LasseRosenow commented on src/platform/posix/tcp_ip_channel.c:
```diff
@@ -486,6 +496,8 @@ void TcpIpChannel_ctor(TcpIpChannel *self, const char *host, unsigned short port
   self->state = NETWORK_CHANNEL_STATE_UNINITIALIZED;
 
+  self->super.get_dest_channel_id = TcpIpChannel_get_dest_channel_id;
+  self->super.get_connection_state = TcpIpChannel_get_connection_state;
```
It seems the `connection_state` mapping was missing so I secretly added it here also

erlingrj replied:
Ops! Thanks!

erlingrj commented on src/platform/posix/tcp_ip_channel.c:
```diff
@@ -469,7 +469,17 @@ static void TcpIpChannel_free(NetworkChannel *untyped_self) {
+  } else {
+    return self->fd;
+  }
+}
```
This depends on how `get_dest_channel_id` is to be used by the runtime. In fact, I think it might not be used by the runtime at all, only by the lower levels of the NetworkChannel implementation. This suggests that it might not be the right decision to have this function as part of the generic NetworkChannel API, rather it some bookkeeping that must be done in the different concrete implementations?

LasseRosenow commented:
Okay I rebased this and removed the `conn_channel_id` from the `NetworkChannel`

The other small change is that I renamed the `get_connection_state` function in the `TcpIpChannel` to match the name of the `NetworkChannel` interface.

erlingrj reviewed (approved):
Great!
